Plugin config file

alxvth opened the pull request:
- Move `src/ScatterplotPlugin.json` to `./PluginInfo.json`
- Add compatible core version to config file
- Read plugin version from config file in cmake
- Append plugin and core version to output library name
   - `ScatterplotPlugin_p1.0.0_c1.4.0.dll` on Windows
   - `libScatterplotPlugin_p1.0.0_c1.4.0.dylib` on Mac
   - `libScatterplotPlugin_p1.0.0_c1.4.0.so` on Linux

Depends on https://github.com/ManiVaultStudio/core/pull/910, https://github.com/ManiVaultStudio/core/pull/911

---

- [x] We could move the snippet that reads the plugin version from the config file into the core and install it as a cmake utility function. Then, it could only be called after `find_package(ManiVault)`, but not each plugin had to repeat those lines.
- ~~I'm not sure about the `core` version entry in the plugin config file. We could keep it as a list, or name it `coreMinVersion` maybe. Either way, the library name will contain the core version that the plugin is build against.~~

ThomasKroes reviewed (commented):
I have a small suggestion: 

{
    "name" : "Scatterplot View",
    "version" :  {
        "plugin": "1.0.0",
        "core": ["1.3"]
    },
    "dependencies" : ["Points"],
}

To me bundling version-related values makes a bit more sense.

alxvth commented:
I moved the config handling to the core, nested the versions and added an optional plugin type field to the config file.

The setup call `mv_handle_plugin_config(${PROJECT})` now prints:
```
-- ManiVault plugin: Scatterplot View
--   version: 1.0.0
--   type: Viewer
```
